src/PureComponent.js

```diff
@@ -0,0 +1,45 @@
+import React, { Component } from "react";
+import { useState } from "react";
+
+const CreateElementComponent = React.createElement(
+  "h2",
+  null,
+  "CreateElementComponent"
+);
+
+class ReactComponent extends Component {
+  render() {
+    return <h2>ReactComponent</h2>;
+  }
+}
+
+const FunctionalComponent = () => {
+  const [count, setCount] = useState(0);
+  const inc = () => {
+    setCount((pv) => pv + 1);
+  };
+  const dec = () => {
+    setCount((pv) => pv - 1);
+  };
+  return (
+    <div>
+      <button onClick={inc}>+</button>
+      <button onClick={dec}>-</button>
+      <br />
+      <h2>{count}</h2>
+    </div>
+  );
+};
+
+function PureComponent() {
```

Comment:
It is not what Pure Component means. Read about React.PureComponent for class components and React.memo for functional

Reply:
`PureComponent implies that there will be no state inside, which is quite logical. "Pure" means "no state".`
I understand that my component is suitable, if not, specify please what the error is. Thanks

Reply:
Not necessarily. In the task we can see it is specifically stated to use React.PureComponent. Its main idea is to reduce the amount of re-renders. PureComponents may have state and props. What you are talking about is more of how pure functions should work in a functional programming paradigm. Yet for React it's more about 'same input - same output'. I asked you to read about React.PureComponent and React.memo because they are two examples how we can make our components pure in React sense. First one implements shouldComponentUpdate() method with shallow comparison of state and props. Second one is used for functional components and implies memoization to prevent unnecessary rerenders.

Reply:
It is true, that your component has no props or state, so it will render its children and that's it, but it's just that the task here is to try out all of the methods to create Components in React, including React.PureComponent or React.memo

Reply:
Okay, ill do it to have 5 stars) thanks

Reply:
It is not necessary, I can set it 5, if you understand what those 2 are for

Reply:
But you should submit this PR to learn, otherwise I can't grade it

Reply:
not so far, still need to read about them. Ill submit to learn portal after ill understand about React.PureComponent and React.memo